packages/languages/src/browser/language-client-contribution.ts

```diff
@@ -87,7 +89,11 @@ export abstract class BaseLanguageClientContribution implements LanguageClientCo
         const fileEvents = this.createFileEvents();
         return {
             documentSelector: this.documentSelector,
-            synchronize: { fileEvents }
+            synchronize: { fileEvents },
+            initializationFailedHandler: () => {
+                this.messageService.error("Failed to start language server '" + this.name + "'.");
+                return false;
```

Comment:
Does it mean don’t try to restart?

Reply:
It still tries to restart. From what I read in the code, the boolean can be set to true in order to call initialize again, perhaps with different options in case of failure. But if set to false, it still gets restarted at a higher level by BaseLanguageClient.handleConnectionClosed

Reply:
Ok, could you extract the function to create `initializationFailedHandler` that clients can change the default behavior, please?